topics/single-cell/tutorials/pseudobulk-analysis/tutorial.md

```diff
@@ -0,0 +1,465 @@
+> >
+> > 3. The **pseudobulk count matrix** is the primary input required for analysis using **edgeR**, a tool designed for differential expression analysis.
+> >    The **Samples Metadata** is another file that will serve as an input for the edgeR tool.
+> >
+> {: .solution}
+>
+{: .question}
+
+## Sanitation Steps - Part 1
+
+The next steps will help you refine your data for easier handling. We will use standard galaxy tools, like: [Replace Text](https://toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1), [Remove columns](https://toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0) and [Text reformatting](https://toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/9.3+galaxy1).
+
+> <hands-on-title>Replace Text and Remove Columns</hands-on-title>
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `count_matrix` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"Find pattern"*: `[ --+*^]+`
+>            - *"Replace with:"*: `_`
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `samples_metadata` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"Find pattern"*: `[ --+*^]+`
+>            - *"Replace with:"*: `_`
+>
+> 1. {% tool [Remove columns](toolshed.g2.bx.psu.edu/repos/iuc/column_remove_by_header/column_remove_by_header/1.0) %} with the following parameters:
+>    - {% icon param-file %} *"Tabular file"*: `genes_metadata` (output of **Decoupler pseudo-bulk** {% icon tool %})
+>    - In *"Select Columns"*:
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `start`
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `end`
+>        - {% icon param-repeat %} *"Insert Select Columns"*
+>            - *"Header name"*: `width`
+>
+> 1. {% tool [Replace Text](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1) %} with the following parameters:
+>    - {% icon param-file %} *"File to process"*: `outfile` (output of **Replace Text: Sample Metadata Step** {% icon tool %})
+>    - In *"Replacement"*:
+>        - {% icon param-repeat %} *"Insert Replacement"*
+>            - *"in column"*: `c2`
+>            - *"Find pattern"*: `^([0-9])(.+)`
+>            - *"Replace with"*: `GG_\\1\\2`
+>
+>    {% snippet faqs/galaxy/analysis_regular_expressions.md %}
+{: .hands_on}
```

Comment:
I think this section will feel like magic to many people, I have added the FAQ about regular expressions at the bottom of this box, but maybe you can also do a comparison of the files before and after these steps? show a couple of lines of the files before and after orso? Or at least add a line per step saying what the tool is doing to the files